### PR TITLE
Upgrade dotnet version to net6 for version controller

### PR DESCRIPTION
**Pull request by BethanyZhou**
## Description
https://dev.azure.com/azclitools/internal/_build/results?buildId=59020&view=results

## Checklist

- [ ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: .azure-pipelines/util/build-steps.yml [hunk 1]
```diff
@@ -35,13 +35,6 @@ steps:
   inputs:
     packageType: sdk
     version: 6.0.x
-
```
Comment (wyunchi-ms) [inline .azure-pipelines/util/build-steps.yml:38]: Please also remove task `Use .NET Core sdk 3.1.x`.

File: tools/StaticAnalysis/StaticAnalysis.Netcore.csproj [hunk 2]
```diff
@@ -1,10 +1,10 @@
-<Project Sdk="Microsoft.NET.Sdk">
+﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <Import Project="$(MSBuildThisFileDirectory)..\..\Repo.props" />
 
   <PropertyGroup>
     <OutputType>Exe</OutputType>
-    <TargetFramework>netcoreapp3.1</TargetFramework>
+    <TargetFramework>net6.0</TargetFramework>
```
Comment (isra-fel) [inline tools/StaticAnalysis/StaticAnalysis.Netcore.csproj:7]: Please investigate whether net6 and net6.0 are the same and make it consistent.

**Review by isra-fel — CHANGES_REQUESTED** [review]
Now that both Az and predictor will be built with .net 6 sdk

File: tools/StaticAnalysis/StaticAnalysis.Netcore.csproj (hunk 2 above)
Comment (isra-fel) [reply tools/StaticAnalysis/StaticAnalysis.Netcore.csproj:7]: According to https://learn.microsoft.com/en-us/dotnet/standard/frameworks#:~:text=6-,net6.0,-2.1 `net6.0` is the formal term.

File: tools/Tools.Common.Test/Tools.Common.Test.csproj [hunk 3]
```diff
@@ -3,7 +3,7 @@
   <Import Project="$(MSBuildThisFileDirectory)..\Common.Netcore.Dependencies.Test.targets" />
 
   <PropertyGroup>
-    <TargetFramework>netcoreapp3.1</TargetFramework>
+    <TargetFramework>net6</TargetFramework>
```
Comment (BethanyZhou) [inline tools/Tools.Common.Test/Tools.Common.Test.csproj:6]:
```suggestion
    <TargetFramework>net6.0</TargetFramework>
```

File: tools/StaticAnalysis/StaticAnalysis.Netcore.csproj (hunk 2 above)
Comment (BethanyZhou) [reply tools/StaticAnalysis/StaticAnalysis.Netcore.csproj:7]: Nice catch, thank you!

**Review by isra-fel — APPROVED** [review]

**Pull request merged**